src/garage/torch/policies/context_conditioned_policy.py

```diff
@@ -0,0 +1,283 @@
+                * np.ndarray: Log std of distribution.
+                * torch.Tensor: Log likelihood of distribution.
+                * torch.Tensor: Sampled values from distribution before
+                    applying tanh transformation.
+            torch.Tensor: z values, with shape :math:`(N, latent_dim)`.
+
+        """
+        self.infer_posterior(context)
+        self.sample_from_belief()
+        task_z = self.z
+
+        # task, batch
+        t, b, _ = obs.size()
+        obs = obs.view(t * b, -1)
+        task_z = [z.repeat(b, 1) for z in task_z]
+        task_z = torch.cat(task_z, dim=0)
+
+        # run policy, get log probs and new actions
+        obs_z = torch.cat([obs, task_z.detach()], dim=1)
+        dist = self._policy(obs_z)
+        pre_tanh, actions = dist.base_dist.rsample_with_pre_tanh_value()
+        log_pi = dist.base_dist.log_prob(value=actions,
+                                         pre_tanh_value=pre_tanh)
+        log_pi = log_pi.unsqueeze(1)
+        mean = dist.mean.to('cpu').detach().numpy()
+        log_std = (dist.variance**.5).log().to('cpu').detach().numpy()
+
+        return (actions, mean, log_std, log_pi, pre_tanh), task_z
+
+    def get_action(self, obs):
+        """Sample action from the policy, conditioned on the task embedding.
+
+        Args:
+            obs (torch.Tensor): Observation values, with shape :math:`(1, O)`.
+                O is the size of the flattened observation space.
+
+        Returns:
+            torch.Tensor: Output action value, with shape :math:`(1, A)`.
+                A is the size of the flattened action space.
+            dict:
+                * np.ndarray[float]: Mean of the distribution.
+                * np.ndarray[float]: Standard deviation of logarithmic values
+                    of the distribution.
+
+        """
+        z = self.z
+        obs = torch.as_tensor(obs[None], device=tu.global_device()).float()
+        obs_in = torch.cat([obs, z], dim=1)
+        action, info = self._policy.get_action(obs_in)
+        action = np.squeeze(action, axis=0)
+        info['mean'] = np.squeeze(info['mean'], axis=0)
+        return action, info
+
+    def compute_kl_div(self):
+        r"""Compute :math:`KL(q(z|c) \| p(z))`.
+
+        Returns:
+            float: :math:`KL(q(z|c) \| p(z))`.
+
+        """
+        prior = torch.distributions.Normal(
+            torch.zeros(self._latent_dim).to(tu.global_device()),
+            torch.ones(self._latent_dim).to(tu.global_device()))
+        posteriors = [
+            torch.distributions.Normal(mu, torch.sqrt(var)) for mu, var in zip(
+                torch.unbind(self.z_means), torch.unbind(self.z_vars))
+        ]
+        kl_divs = [
+            torch.distributions.kl.kl_divergence(post, prior)
+            for post in posteriors
+        ]
+        kl_div_sum = torch.sum(torch.stack(kl_divs))
+        return kl_div_sum
+
+    def log_diagnostics(self, eval_statistics):
```

Comment:
this style of log diagnostics API isn't used in garage -- instead we log directly from log diagnostics.

Your log should have a prefix with the name of your primitive, i.e. the info is logged to `Name/ZMeanEval`, etc.

The default name should be the class name, but you should provide users the option (constructor argument) to name the primitive something different (self._name).